File: 0965-univalued-binary-tree/0965-univalued-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public int val;
    public boolean compute(TreeNode root){
        if(root==null)return true;
        return root.val==val && compute(root.left) && compute(root.right);
    }
    public boolean isUnivalTree(TreeNode root) {
        if(root==null)return true;
        val = root.val;
        return compute(root);
        
    }
}

File: 0965-univalued-binary-tree/0965-univalued-binary-tree.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def isUnivalTree(self, root: Optional[TreeNode]) -> bool:
        if not root : return True
        val = root.val
        def compute(root):
            if not root:
                return True
            return root.val==val and compute(root.left) and compute(root.right)
        return compute(root)
        